images/linux/scripts/installers/docker.sh

```diff
@@ -1,25 +1,26 @@
 #!/bin/bash -e
 ################################################################################
-##  File:  docker-moby.sh
+##  File:  docker.sh
 ##  Desc:  Installs docker onto the image
 ################################################################################
 
 # Source the helpers for use with the script
+source $HELPER_SCRIPTS/os.sh
 source $HELPER_SCRIPTS/install.sh
 
-# Check to see if docker is already installed
-docker_package=moby
-echo "Determing if Docker ($docker_package) is installed"
-if ! IsPackageInstalled $docker_package; then
-    echo "Docker ($docker_package) was not found. Installing..."
-    apt-get remove -y moby-engine moby-cli
-    apt-get update
-    apt-get install -y moby-engine moby-cli
-    apt-get install --no-install-recommends -y moby-buildx
-    apt-get install -y moby-compose
-else
-    echo "Docker ($docker_package) is already installed"
-fi
+repo_url="https://download.docker.com/linux/ubuntu"
+gpg_key="/usr/share/keyrings/docker.gpg"
+repo_path="/etc/apt/sources.list.d/docker.list"
+
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o $gpg_key
+echo "deb [arch=amd64 signed-by=$gpg_key] $repo_url $(getOSVersionLabel) stable" > $repo_path
+apt-get update
+apt-get install --no-install-recommends docker-ce docker-ce-cli containerd.io docker-buildx-plugin
+
+# Install docker compose v2 from releases
+URL=$(get_github_package_download_url "docker/compose" "contains(\"compose-linux-x86_64\")")
+curl -fsSL $URL -o /usr/libexec/docker/cli-plugins/docker-compose
+chmod +x /usr/libexec/docker/cli-plugins/docker-compose
```

Comment:
Why not install docker-compose from the apt repository? I'm asking as I'm seeing failures in the [Dokku project CI](https://github.com/dokku/dokku) after the runner-images update (we're failing to upgrade docker-compose).

Reply:
Docker compose packages in both `packages.microsoft.com` and `download.docker.com` tend to lag behind the version from github releases, resulting in fixes arriving late